src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1120,12 +1119,12 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 						shellLaunchConfig.args = undefined;
 					}
 					shellLaunchConfig.executable = await this._resolveVariable(variableResolver, shellOptions.executable);
-					shellSpecified = true;
 				}
 				if (shellOptions.args) {
 					shellLaunchConfig.args = await this._resolveVariables(variableResolver, shellOptions.args.slice());
 				}
 			}
+			const taskShellArgsSpecified = shellOptions?.args !== undefined;
```

Comment:
The difference between #181760 is that we check for `shellOptions?.args` instead of `shellLaunchConfig.args` here, so that the default profile's args don't affect us from adding in args like `-c`